### ISSUE TITLE
Support serialization and deserialization of unresolved backtraces

### ISSUE THREAD

**Issue by helgoboss** [rust-lang/backtrace-rs#353]
Background: I want to deliver a binary (a cdylib) without debug symbols, for size reasons. If there's a panic and a user sends me a backtrace, it contains almost no usable information for debugging the issue. I was thinking about core dumps but can't use them because aborting the program just because of a panic is a bad option (it's a plug-in and I don't want to tear down the complete host application).

I was hoping that this could be a way:

1. Create an unresolved backtrace via `Backtrace::new_unresolved()` (I guess that wouldn't need debug symbols, e.g. PDB, to be available)
2. Serialize it into a file
3. Let the user send me that file
4. I load it into a small backtrace inspection tool
5. This tool deserializes the unresolved backtrace, calls `resolve()` (works because I have the debug symbols) and formats it.

Does this make sense? Or do you think there's a better way to get good crash reports without shipping huge binaries?

**Comment by alexcrichton** [comment]
Unfortunately this library may not help a lot with this use case. While what you're describing is definitely possible (and I know various companies that do that) this crate isn't really targeted at this use case. This crate is targeted at capturing and printing backtraces in-process, not across process boundaries (or for remote processes like those already exited from before).

I think this could be implemented (or may work already?) as-described, but it may not help you achieve your goal.

**Comment by helgoboss** [comment]
@alexcrichton Thanks for your response! I guessed already that this might be a bit out of scope for this crate.

For anyone else confronted with the same situation: Here's a little [Gist](https://gist.github.com/helgoboss/1c4f9ef260f5001cd85fbad53d337809) I came up with for resolving symbols in the binary itself, assuming that both user and developer use the very same binary and the developer still has access to the debug symbols belonging to that binary. Although not very comfortable, so far it seems to work and it doesn't need any third-party tools. Just tested it on Windows, but I think it should work cross-platform (because backtrace-rs abstracts away the native symbol resolution stuff - very grateful for that 😃).

Closing this now. If anybody knows a better way to achieve this goal, I would be interested in knowing about it (preferably cross-platform and without commercial tooling).
